metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -158,113 +301,144 @@ def __init__(self, config: BigQueryV2Config, report: BigQueryV2Report):
             except Exception as e:
                 self.report.usage_failed_extraction.append(project_id)
+                self.report.report_failure("usage-extraction", str(e))
                 trace = traceback.format_exc()
                 logger.error(
-                    f"Error getting usage for project {project_id} due to error {e}, trace: {trace}"
+                    f"Error getting usage events for project {project_id} due to error {e}, trace: {trace}"
                 )
 
             self.report.usage_extraction_sec[project_id] = round(
                 timer.elapsed_seconds(), 2
             )
 
-    def _get_bigquery_log_entries_via_exported_bigquery_audit_metadata(
-        self, client: BigQueryClient
-    ) -> Iterable[BigQueryAuditMetadata]:
-        try:
-            list_entries: Iterable[
-                BigQueryAuditMetadata
-            ] = self._get_exported_bigquery_audit_metadata(
-                client, self.config.get_table_pattern(self.config.table_pattern.allow)
-            )
-            i: int = 0
-            for i, entry in enumerate(list_entries):
-                if i == 0:
-                    logger.info(
-                        f"Starting log load from BigQuery for project {client.project}"
-                    )
-                yield entry
-
-            logger.info(
-                f"Finished loading {i} log entries from BigQuery for project {client.project}"
-            )
-
-        except Exception as e:
-            logger.warning(
-                f"Encountered exception retrieving AuditLogEntries for project {client.project} - {e}"
-            )
-            self.report.report_failure(
-                "lineage-extraction",
-                f"{client.project} - unable to retrieve log entries {e}",
-            )
+    def _store_usage_event(
+        self,
+        event: AuditEvent,
+        usage_state: BigQueryUsageState,
+        table_refs: Set[str],
+    ) -> bool:
+        """Stores a usage event in `usage_state` and returns if an event was successfully processed."""
+        if event.read_event and (
+            self.config.start_time <= event.read_event.timestamp < self.config.end_time
+        ):
+            resource = event.read_event.resource
+            if str(resource) not in table_refs:
+                logger.debug(f"Skipping non existing {resource} from usage")
+                return False
+            elif resource.is_temporary_table([self.config.temp_table_dataset_prefix]):
+                logger.debug(f"Dropping temporary table {resource}")
+                self.report.report_dropped(str(resource))
+                return False
+
+            # Use uuid keys to store all entries -- no overwriting
+            key = str(uuid.uuid4())
+            usage_state.read_events[key] = event.read_event
+            for field in event.read_event.fieldsRead:
+                usage_state.column_accesses[str(uuid.uuid4())] = key, field
```

Comment:
Why can't we store this alongside the read_events?

Reply:
That'd require storing a list in a single column right? I can store it as a blob but then I can't write sqlite queries against it, right? e.g. https://stackoverflow.com/questions/1517771/is-it-possible-to-save-a-list-of-values-into-a-sqlite-column